src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -548,9 +545,51 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
+        private Dictionary<string, ConfigurationSetting> ReorderData(Dictionary<KeyValueIdentifier, string> cachedInfoLogs = default)
+        {
+            Dictionary<string, ConfigurationSetting> orderedData = new Dictionary<string, ConfigurationSetting>();
+            Dictionary<string, string> logIdentifiers = new Dictionary<string, string>();
+
```

Comment:
Doesn't this break simple label override scenario?

```
options.Connect()
    .Select(KeyFilter.All, LabelFilter.Null)
    .Select(KeyFilter.All, "Prod")
```

Reply:
Since I used a `Dictionary<TKey, TValue>` this scenario could break when the values in the dictionary change, so I could use an `OrderedDictionary` instead. However, `OrderedDictionary` doesn't let you specify types like `<TKey, TValue>` when creating one, which is a drawback. I'm still trying to find a better way to solve this.

Reply:
https://github.com/dotnet/runtime/issues/24826